tests/test_dragon_launcher.py

```diff
@@ -0,0 +1,75 @@
+import typing as t
+
+import pytest
+
+from smartsim._core.launcher.dragon.dragonLauncher import DragonLauncher
+from smartsim._core.schemas.dragonRequests import DragonBootstrapRequest
+from smartsim.error.errors import LauncherError
+
+
+class MockPopen:
+    def __init__(self, *args: t.Any, **kwargs: t.Any) -> None: ...
+
+    @property
+    def pid(self) -> int:
+        return 1
+
+    @property
+    def returncode(self) -> int:
+        return 0
+
+
+class MockSocket:
+    def __init__(self, *args: t.Any, **kwargs: t.Any) -> None:
+        self._bind_address = ""
+
+    def __call__(self, *args: t.Any, **kwds: t.Any) -> t.Any:
+        return self
+
+    def bind(self, addr: str) -> None:
+        self._bind_address = addr
+
+    def recv_json(self) -> str:
+        dbr = DragonBootstrapRequest(address=self._bind_address)
+        return dbr.json()
+
+    def close(self) -> None: ...
+
+    def send_json(self, json: str) -> None: ...
+
+    @property
+    def bind_address(self) -> str:
+        return self._bind_address
+
+
+def test_dragon_connect_bind_address(monkeypatch: pytest.MonkeyPatch, test_dir: str):
+    """Test the connection to a dragon environment dynamically selects an open port
+    in the range supplied"""
+
+    with monkeypatch.context() as ctx:
+        ctx.setenv("SMARTSIM_DRAGON_SERVER_PATH", test_dir)
+        ctx.setattr(
+            "smartsim._core.launcher.dragon.dragonLauncher.get_best_interface_and_address",
+            lambda: ("faux_interface", "127.0.0.1"),
+        )
+        ctx.setattr(
+            "smartsim._core.launcher.dragon.dragonLauncher.DragonLauncher._handshake",
+            lambda self, address: ...,
+        )
+        ctx.setattr(
+            "smartsim._core.schemas.utils.SchemaSerializer.mapping_to_schema",
```

Comment:
Is this class/method still around after #8? I'm a little surprised this is not error-ing out, hahaha

Reply:
If you've merged, I'll need to just grab latest and tweak...

Reply:
fixed